app/models/product.rb

```diff
@@ -6,6 +6,11 @@ class Product < ApplicationRecord
   has_many :orders, through: :line_items
   before_destroy :ensure_not_referenced_by_any_line_item
 
+  after_initialize do |prod| 
```

Comment:
When is after_initialize invoked??
This is error prone. Please fix

Reply:
Done